kratos/processes/sub_model_part_skin_detection_process.h

```diff
@@ -0,0 +1,232 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+
+#if !defined(KRATOS_SUB_MODEL_PART_SKIN_DETECTION_PROCESS_H_INCLUDED)
+#define KRATOS_SUB_MODEL_PART_SKIN_DETECTION_PROCESS_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+
+// Project includes
+#include "skin_detection_process.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+template<SizeType TDim>
+class SubModelPartSkinDetectionProcess: public SkinDetectionProcess<TDim>
+{
+
+KRATOS_DEFINE_LOCAL_FLAG( NODE_SELECTED );
+
+class FaceSelector
+{
+public:
+KRATOS_CLASS_POINTER_DEFINITION(FaceSelector);
+virtual void Prepare(ModelPart& rMainModelPart) const = 0;
+virtual bool IsSelected(const Geometry<Node<3>>::PointsArrayType&) const = 0;
+};
+
+class SelectIfAllNodesOnSubModelPart: public FaceSelector
+{
+std::string mName;
+public:
+SelectIfAllNodesOnSubModelPart(const std::string& rName): mName(rName) {}
+
+void Prepare(ModelPart& rMainModelPart) const override
```

Comment:
No Doxygen?

Reply:
I'll add some doxygen for the new functions, but these are private classes that cannot be accessed from outside this utility. I'll add some normal c++ comments to explain it, but I'd prefer it if they do not appear in doxygen, which should be a documentation for "users" of the class.